pyccl/covariances.py

```diff
@@ -154,35 +157,40 @@ def sigma2_B_disc(cosmo, a=None, fsky=1., p_of_k_a=None):
             power spectrum to use. Defaults to `None`, in which case the
             internal linear power spectrum from `cosmo` is used.
 
+
     Returns:
-        float or array_like: values of the projected variance.
+        a_arr (array_like): an array of scale factor values at which the
+            projected variance has been evaluated. Only returned if `a_arr` is
+            `None` on input.
+        sigma2_B (float or array_like): projected variance.
     """
-    status = 0
-    if a is None:
-        na = lib.get_pk_spline_na(cosmo.cosmo)
-        a_arr, status = lib.get_pk_spline_a(cosmo.cosmo, na, status)
-        check(status, cosmo=cosmo)
+    full_output = a_arr is None
+
+    if full_output:
+        a_arr = cosmo.get_pk_spline_a()
+        ndim = np.ndim(a_arr)
```

Comment:
This `ndim` isn't used anywhere because the function returns if `full_output is True`.

Reply:
Yep, done